superset/connectors/sqla/models.py

```diff
@@ -141,19 +157,11 @@ def get_timestamp_expression(self, time_grain):
         is_epoch = pdf in ('epoch_s', 'epoch_ms')
         if not self.expression and not time_grain and not is_epoch:
             return column(self.column_name, type_=DateTime).label(label)
-
-        expr = self.expression or self.column_name
-        if is_epoch:
-            # if epoch, translate to DATE using db specific conf
-            db_spec = self.table.database.db_engine_spec
-            if pdf == 'epoch_s':
-                expr = db_spec.epoch_to_dttm().format(col=expr)
-            elif pdf == 'epoch_ms':
-                expr = db_spec.epoch_ms_to_dttm().format(col=expr)
-        if time_grain:
-            grain = self.table.database.grains_dict().get(time_grain)
-            if grain:
-                expr = grain.function.format(col=expr)
+        grain = self.table.database.grains_dict().get(time_grain) if time_grain else None
+        expr = self.db_spec.get_time_expr(self.expression or self.column_name,
+                                          pdf,
+                                          time_grain,
+                                          grain)
```

Comment:
Moving this to `db_engine_specs` is probably a good thing even in the absence of pinot, reducing some unnecessary clutter. Perhaps compact pdf, time_grain and grain to the same row?

Reply:
Good idea.